flink-runtime/src/main/java/org/apache/flink/runtime/io/network/netty/NetworkBufferAllocator.java

```diff
@@ -69,6 +69,9 @@ Buffer allocatePooledNetworkBuffer(InputChannelID receiverId) {
      * @return The un-pooled network buffer.
      */
     Buffer allocateUnPooledNetworkBuffer(int size, Buffer.DataType dataType) {
+        if (size <= 0) {
+            return null;
+        }
```

Comment:
1. `@Nullable`
2. Do we really need to support `null` here? Can not we return empty `Buffer`?

Reply:
This method is used to allocate unpooled buffer for event, the size must be positive. I will replace it with a argument check.